Can't run on MacOS

Works fine on Linux, but on MacOS, running magneticod I get:
```
$ magneticod 
2020-02-04T16:17:06.938+1100	INFO	magneticod v0.9.0 has been started.
2020-02-04T16:17:06.939+1100	INFO	Copyright (C) 2017-2019  Mert Bora ALPER <bora@boramalper.org>.
2020-02-04T16:17:06.939+1100	INFO	Dedicated to Cemile Binay, in whose hands I thrived.
2020-02-04T16:17:06.939+1100	INFO	Compiled on 2020-02-04T05:17:00Z
2020-02-04T16:17:06.939+1100	FATAL	Could not open the database at `sqlite3:///Users/anacrolix/Library/Application Support/magneticod/database.sqlite3?_journal_mode=WAL&_busy_timeout=3000&_foreign_keys=true`%!(EXTRA zapcore.Field={error 25 0  sql.DB.Ping: unable to open database file})
```

Same error on `master` and `v0.9.0`.

There's more:
```
--- FAIL: TestAppdirs (0.00s)
    main_test.go:15: UserDataDir returned an unexpected value!  `/Users/anacrolix/Library/Application Support/magneticod`
    main_test.go:21: UserCacheDir returned an unexpected value!  `/Users/anacrolix/Library/Caches/magneticod`
FAIL
```

--- a/pkg/persistence/sqlite3.go
+++ b/pkg/persistence/sqlite3.go
@@ -31,7 +31,11 @@ func makeSqlite3Database(url_ *url.URL) (Database, error) {
 	}
 
 	var err error
-	url_.Scheme = ""
+	// To handle spaces in the file path, we ensure that URI path handling is triggered in the
+	// sqlite3 driver, and that escaping is applied to the URL on this side. See issue #240.
+	url_.Scheme = "file"
+	// To ensure that // isn't injected into the URI. The query is still handled.
+	url_.Opaque=url_.Path
 	db.conn, err = sql.Open("sqlite3", url_.String())
 	if err != nil {
 		return nil, errors.Wrap(err, "sql.Open")

